src/MaterialDesignThemes.Wpf/DialogHost.cs

```diff
@@ -405,6 +409,50 @@ private static void IsOpenPropertyChangedCallback(DependencyObject dependencyObj
         })));
     }
 
+    
+    private void ListenForWindowStateChanged(Window? window)
+    {
+        window ??= Window.GetWindow(this);
+
+        if (window is not null)
+        {
+            window.StateChanged += Window_StateChanged;
```

Comment:
We want to make sure we don't register twice.
```suggestion
            window.StateChanged -= Window_StateChanged;
            window.StateChanged += Window_StateChanged;
```